plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProject.java

```diff
@@ -47,9 +50,14 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
     CreateProjectOperation operation = new CreateProjectOperation(
         description, "Creating new App Engine Project");
     try {
-      operation.execute(progress.newChild(50), uiInfoAdapter);
-      
-      CodeTemplates.materialize(newProject, config, progress.newChild(50));
+      operation.execute(progress.newChild(20), uiInfoAdapter);
+      IFacetedProject facetedProject = ProjectFacetsManager.create(
+          newProject, true, progress.newChild(40));
+      IProjectFacet javaFacet = ProjectFacetsManager.getProjectFacet("jst.java");
```

Comment:
So it turns out `jst.java` is a facet alias to `java`, so we can avoid some of this lookup here by using `JavaFacet.VERSION_1_7`.